wolfcrypt/src/random.c

```diff
@@ -198,58 +209,73 @@ static int Hash_df(DRBG* drbg, byte* out, word32 outSz, byte type,
                                                   const byte* inA, word32 inASz,
                                                   const byte* inB, word32 inBSz)
 {
+    int ret;
     byte ctr;
     int i;
     int len;
     word32 bits = (outSz * 8); /* reverse byte order */
     Sha256 sha;
-    byte digest[SHA256_DIGEST_SIZE];
+    DECLARE_VAR(digest, byte, SHA256_DIGEST_SIZE, drbg->heap);
```

Comment:
I agree that's not obvious that it can do a malloc here with the async. The macros help improve performance for variables that need allocated for hardware using (NUMA). This macro is used in a few places in the code and wolfCrypt test/benchmark. The macro is defined in types.h. @toddouska : what are your thoughts on this?